base/applications/drwtsn32/sysinfo.cpp

```diff
@@ -71,6 +73,38 @@ void PrintSystemInfo(FILE* output, DumpData& data)
         xfprintf(output, "    User Name: %s" NEWLINE, Buffer);
 
 
+    PVOID SMBiosBuf;
+    PCHAR DmiStrings[ID_STRINGS_MAX] = { 0 };
+    SMBiosBuf = LoadSMBiosData(DmiStrings);
+    if (SMBiosBuf)
+    {
+        if (DmiStrings[BIOS_VENDOR])
+            xfprintf(output, "    BIOS Vendor: %s" NEWLINE, DmiStrings[BIOS_VENDOR]);
+        if (DmiStrings[BIOS_VERSION])
+            xfprintf(output, "    BIOS Version: %s" NEWLINE, DmiStrings[BIOS_VERSION]);
+        if (DmiStrings[BIOS_DATE])
+            xfprintf(output, "    BIOS Date: %s" NEWLINE, DmiStrings[BIOS_DATE]);
+        if (DmiStrings[SYS_VENDOR])
+            xfprintf(output, "    System Manufacturer: %s" NEWLINE, DmiStrings[SYS_VENDOR]);
+        if (DmiStrings[SYS_FAMILY])
+            xfprintf(output, "    System Family: %s" NEWLINE, DmiStrings[SYS_FAMILY]);
+        if (DmiStrings[SYS_PRODUCT])
+            xfprintf(output, "    System Model: %s" NEWLINE, DmiStrings[SYS_PRODUCT]);
+        if (DmiStrings[SYS_VERSION])
+            xfprintf(output, "    System Version: %s" NEWLINE, DmiStrings[SYS_VERSION]);
+        if (DmiStrings[SYS_SKU])
+            xfprintf(output, "    System SKU: %s" NEWLINE, DmiStrings[SYS_SKU]);
+        if (DmiStrings[BOARD_VENDOR])
+            xfprintf(output, "    Baseboard Manufacturer: %s" NEWLINE, DmiStrings[BOARD_VENDOR]);
+        if (DmiStrings[BOARD_NAME])
+            xfprintf(output, "    Baseboard Model: %s" NEWLINE, DmiStrings[BOARD_NAME]);
+        if (DmiStrings[BOARD_VERSION])
+            xfprintf(output, "    Baseboard Version: %s" NEWLINE, DmiStrings[BOARD_VERSION]);
+        if (DmiStrings[BOARD_ASSET_TAG])
+            xfprintf(output, "    Baseboard Asset Tag: %s" NEWLINE, DmiStrings[BOARD_ASSET_TAG]);
+        FreeSMBiosData(SMBiosBuf);
+    }
```

Comment:
Are all these strings usefull for diagnosing a problem?

Reply:
I think yes. :slightly_smiling_face:  It would allow to know more about target system and its quirks.